mantis-network/src/main/java/io/reactivex/mantis/network/push/TimedChunker.java

```diff
@@ -43,26 +54,67 @@ public TimedChunker(MonitoredQueue<T> buffer, int maxBufferLength,
         this.buffer = buffer;
         this.processor = processor;
         this.connectionManager = connectionManager;
+
+        MetricGroupId metricsGroup = new MetricGroupId("TimedChunker");
+        Metrics metrics = new Metrics.Builder()
+                .id(metricsGroup)
+                .addCounter("chunkerCancelSuccess_interrupted")
+                .addCounter("chunkerCancelFailure_interrupted")
+                .addCounter("chunkerCancelSuccess_timeout")
+                .addCounter("chunkerCancelFailure_timeout")
+                .addCounter("chunkerCancelSuccess_execution")
+                .addCounter("chunkerCancelFailure_execution")
+                .build();
+        chunkerCancelSuccessInterrupted = metrics.getCounter("chunkerCancelSuccess_interrupted");
+        chunkerCancelFailureInterrupted = metrics.getCounter("chunkerCancelFailure_interrupted");
+        chunkerCancelSuccessTimeout = metrics.getCounter("chunkerCancelSuccess_timeout");
+        chunkerCancelFailureTimeout = metrics.getCounter("chunkerCancelFailure_timeout");
+        chunkerCancelSuccessExecution = metrics.getCounter("chunkerCancelSuccess_execution");
+        chunkerCancelFailureExecution = metrics.getCounter("chunkerCancelFailure_execution");
+        MetricsRegistry.getInstance().registerAndGet(metrics);
```

Comment:
Instead of defining one counter per failure type, would it make sense to tag the failure type along with the cancel counter? That way, a lot of the logic below can be generalized irrespective of the failure type?